.buildkite/pipeline.yml

```diff
@@ -0,0 +1,22 @@
+steps:
+  - command: |
+      . /var/lib/buildkite-agent/.nix-profile/etc/profile.d/nix.sh
```

Comment:
This is very specific to how our agent is set up - I'd rather see us add a `nix=true` tag to all nix-enabled agents (which set this locally in an environment hook), and then [require that tag to be present for that specific step](https://buildkite.com/docs/agent/v3/cli-start#agent-targeting).

Reply:
I don't see this as a problem for now: the agent such that it is setup currently is temporary. We'll want to reprovision it as completely immutable infrastructure (not in scope for this PR).